packages/ripple/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import { getRippleGlobalNamespace } from './getRippleGlobalNamespace';
+import { Variant, colorMap, Mode } from './utils';
+
+const TRANSITION_TIME = 300;
+const RIPPLE_NAMESPACE = getRippleGlobalNamespace();
+
+interface Options {
+  variant: Variant;
+  darkMode: boolean;
+}
+
+let globalOptions: Options = { darkMode: false, variant: Variant.Default };
+
+function rippleEvent(event: MouseEvent) {
+  if (
+    // @ts-expect-error using HTMLElements to index as it provides a faster lookup when deciding if we should create a ripple effect on a given element
+    RIPPLE_NAMESPACE.registeredRippleElements[event.target]
+  ) {
+    createRippleEffect(event);
+  }
+}
+
+export function registerRipple(node: HTMLElement, options: Options) {
+  if (!RIPPLE_NAMESPACE) {
```

Comment:
This is implicitly SSR safe due to this. Should we be more explicit about handling SSR?

Reply:
I checked and this does work in our Next app, but added a comment to explain that it was explicitly for handling SSR